apps/server/src/modules/tool/external-tool/uc/external-tool.uc.ts

```diff
@@ -74,6 +80,13 @@ export class ExternalToolUc {
 		return promise;
 	}
 
+	async getMetadataForExternalTool(userId: EntityId, toolId: EntityId): Promise<ExternalToolMetadata> {
+		await this.ensurePermission(userId, Permission.TOOL_ADMIN);
+
+		const metadata: ExternalToolMetadata = await this.externaltoolMetadataService.getMetaData(toolId);
+		return metadata;
+	}
+
 	private async ensurePermission(userId: EntityId, permission: Permission) {
 		const user: User = await this.authorizationService.getUserWithPermissions(userId);
 		this.authorizationService.checkAllPermissions(user, [permission]);
```

Comment:
this should be this.authorizationService.checkPermission and check if you have access to the external tool

Reply:
as discussed this is marked as TODO